database-couchdb/index.js

```diff
@@ -1,103 +1,100 @@
-
-        count++;
-        if (entry.itemId === undefined) {
-          entry.itemId = entry._id = curId.toString();
-          entry.itemAvailability = stores;
-        }
+exports.getData = (itemId) => {
 
-        entry.itemAvailability[parseInt(data.store_id) - 1].availability = data.availability;
-        if (count === 5) {
-          docs.push(entry);
-          entry = {};
-          curId++;
-          count = 0;
+  return axios.post(url + '/_find',
+    {
+      'selector': {
+        _id: {
+          '$eq': itemId
         }
-        if (docs.length === batchSize) {
-
-          console.log(curId);
-          availabilityStream.pause();
-
-          Axios.post(url + db + '/_bulk_docs', { docs })
-            .then(() => {
-              console.log('posted!');
-              setTimeout(() => {
-                console.log('Now data will start flowing again. ' + curId);
-
-                availabilityStream.resume();
-              }, 500);
-            })
-            .catch((err) => {
-              console.error(err);
-            });
-          docs = [];
-        }
-      })
-      .on('end', () => {
-        console.log('seed done!');
-        availabilityStream.destroy();
-      });
-  }
-  getStoreData((stores) => {
-    writeData(curId, stores);
-  });
+      },
+      'fields': ['itemId', 'itemAvailability'],
+      'limit': 1,
+    }
+  )
+    .then((data) => {
+      return data.data.docs[0];
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+};
 
-}
+exports.updateData = async (itemId, info) => {
 
-const getStoreData = (CB) => {
-  let storesStream = fs.createReadStream(storesPath);
-  let stores = [];
-  storesStream
-    .on('error', (err) => {
-      console.error(err);
-    })
-    .pipe(csvParser())
-    .on('data', (data) => {
-      let store = data;
-      store.storeId = data.store_id;
-      store.availability = false;
-      // console.log(store);
-      stores.push(store);
+  info._rev = await getRevId(itemId);
+
+  return axios.put(url + `/${itemId}`, info)
+    .then((data) => {
+      return data.data;
     })
-    .on('end', () => {
-      storesStream.destroy();
-      CB(stores);
+    .catch((err) => {
+      console.error(err);
     });
+};
+
+exports.deleteData = async (itemId) => {
 
+  let rev = await getRevId(itemId);
+
+  return axios.delete(url + `/${itemId}?rev=${rev}`)
+    .then((data) => {
+      return data.data;
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 };
 
-exports.removeOldData = removeOldData;
-exports.createNewData = createNewData;
\ No newline at end of file
+exports.addData = async (data) => {
+
+  //couchDB does not support auto increment indexes so I have to get the size of the DB and
+  //manually assign the new index
+  let id = await axios.get(url)
+    .then((data) => {
+      return data.data.doc_count;
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+
+  return axios.put(url + `/${id + startId}`, data)
```

Comment:
So your waiting for the function above to run before you use this function?

Reply:
Yea, not having auto increment really sucks haha